web/src/lib/utils/file-uploader.ts

```diff
@@ -60,9 +60,21 @@ export const openFileUploadDialog = (uploadType: UploadType) => {
 				albumUploadAssetStore.count.set(acceptedFile.length);
 			}
 
-			for (const asset of acceptedFile) {
-				await fileUploader(asset, uploadType);
-			}
+			await mergeMap(
```

Comment:
Is this the algorithm that you said that can prevent crashing the browser? Can you in some words, explain what does the `mergeMap` method do?

Reply:
I borrow idea from my favorite lib `rxjs.mergeMap`. The `mergeMap` will execute fileUpload function in batch (`concurrent` param is batch size). We will call `fileUploader` function n times at a time if we have n files. To prevent this, I create `mergeMap` to limit the number of calls of the function `fileUploader` to `concurrent` times. We only call `fileUploader` function `concurrent` times even though we have n files. The default `concurrent` is 2. Therefore, we can reduce memory usage significantly.